drivers/sensor/bmi088/bmi088_acc.c

```diff
@@ -0,0 +1,321 @@
+            .val2=converted_val % 1000000};
+    return val;
+}
+
+/**
+ * Convert the corresponding value of the channel (X, Y or Z)
+ *
+ * @param chan Channel to read. Implemented: Acc X, Y, Z
+ * @param scale Value to scale the raw_val
+ * @param raw_xyz array for storing X, Y, Z channel raw value of the sensor
+ * @return
+ */
+struct sensor_value bmi088_acc_channel_convert(enum sensor_channel chan, uint16_t scale, int16_t raw_xyz[3]) {
+    switch (chan) {
+        case SENSOR_CHAN_ACCEL_X:
+            return bmi088_acc_to_fixed_point(raw_xyz[0], scale);
+        case SENSOR_CHAN_ACCEL_Y:
+            return bmi088_acc_to_fixed_point(raw_xyz[1], scale);
+        case SENSOR_CHAN_ACCEL_Z:
+            return bmi088_acc_to_fixed_point(raw_xyz[2], scale);
+        default:
+            LOG_ERR("Channels not supported !");
+            struct sensor_value empty = {0};
+            return empty;
+    }
+}
+
+static int bmi088_acc_attr_set(const struct device *dev, enum sensor_channel chan, enum sensor_attribute attr,
+                           const struct sensor_value *val) {
+    return -ENOTSUP;
+}
+
+/**
+ * API function to retrieve a measurement from the sensor.
+ *
+ * @param dev Sensor device pointer
+ * @param chan Channel to fetch. Only SENSOR_CHAN_ALL and ACCEL_XYZ is supported.
+ * @return 0 on success
+ */
+// TODO: check (5.3.3)
+static int bmi088_acc_sample_fetch(const struct device *dev, enum sensor_channel chan) {
+    struct bmi088_acc_data *data = to_data(dev);
+
+    __ASSERT(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_ACCEL_XYZ, "channel is not valid");
+
+    uint8_t status = 0;
+    while ((status & BMI088_ACC_STATUS_MASK) == 0) {
+
+        if (bmi088_acc_byte_read(dev, BMI088_ACC_STATUS, &status) < 0) {
+            return -EIO;
+        }
+    }
+
+    if (bmi088_acc_read(dev, RATE_X_LSB, data->sample.acc, BMI088_SAMPLE_SIZE) < 0) {
+        return -EIO;
+    }
+
+    // convert samples to cpu endianness
+    for (size_t i = 0; i < BMI088_AXES; i++) {
+        uint16_t *sample = &data->sample.acc[i];
+
+        *sample = sys_le16_to_cpu(*sample);
+    }
+
+    LOG_INF("Fetched %d %d %d", data->sample.acc[0], data->sample.acc[1], data->sample.acc[2]);
+
+    return 0;
+}
+
+/**
+ * API function to get a cached sensor value that was previously fetched from the sensor
+ *
+ * @param dev Sensor device pointer
+ * @param chan Channel to read. Implemented: Acc X, Y, Z
+ * @param [out] val Sensor value
+ * @return 0 on success, -ENOTSUP on unsupported channel
+ */
+static int bmi088_acc_channel_get(const struct device *dev, enum sensor_channel chan, struct sensor_value *val) {
+
+    uint8_t range_value = 0;
+    // reading ACC_RANGE for calculation of the scale below
+    if (bmi088_acc_byte_read(dev, ACC_RANGE, &range_value) < 1) {
+        return -EIO;
+    }
+    // scale for converting sensor output to mg (Datasheet 5.3.4)
+    const uint16_t scale = (32768 * 1000 * pow(2,(range_value + 1)) * 1.5);
```

Comment:
```suggestion
    const uint16_t scale = (32768 * 1000 * (1 << (range_value + 1)) * 1.5);
```

ja nicht schön und so aber `pow()` gibts wohl nur für floating point...

Reply:
ah ja wo kommt eigentlich das ` * 1.5` her?